observing/makesdf.py

```diff
@@ -1,150 +1,135 @@
 import pandas as pd
 import numpy as np
 from astropy.time import Time
-import warnings
 from datetime import timedelta
-import subprocess
-import argparse
+import random
+import os
+import logging
+
+logger = logging.getLogger('observing')
 
-MODES = ['POWER', 'VOLT', 'FAST', 'SLOW']
 
+MODES = ['POWER', 'VOLT', 'FAST', 'SLOW']
```

Comment:
I haven't used `enum` but `classes.py` includes one for these modes. The issue is that this list is mostly for comparison (e.g., `if name in MODES...`) and it seems like an `enum` doesn't make that easy.

Reply:
The point is kinda not to have list of strings scattered everywhere. When you check for obs mode, you almost certainly want to throw an error when it's not a valid ObsMode? You can just go ahead and typecast (I think you can also add another constructor to the Enum class that converts to upper/lower case)
```
In [3]: class ObsMode(Enum):
   ...:     Power='POWER'
   ...:     Volt='VOLT'
   ...: 

In [4]: arg = 'voLt'

In [5]: ObsMode(arg.upper())
Out[5]: <ObsMode.Volt: 'VOLT'>

In [6]: ObsMode('Penguin')
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Input In [6], in <cell line: 1>()
----> 1 ObsMode('Penguin')

File /opt/miniconda3/lib/python3.9/enum.py:384, in EnumMeta.__call__(cls, value, names, module, qualname, type, start)
    359 """
    360 Either returns an existing member, or creates a new enum class.
    361 
   (...)
    381 `type`, if set, will be mixed in as the first base class.
    382 """
    383 if names is None:  # simple value lookup
--> 384     return cls.__new__(cls, value)
    385 # otherwise, functional API: we're creating a new Enum type
    386 return cls._create_(
    387         value,
    388         names,
   (...)
    392         start=start,
    393         )

File /opt/miniconda3/lib/python3.9/enum.py:702, in Enum.__new__(cls, value)
    700 ve_exc = ValueError("%r is not a valid %s" % (value, cls.__qualname__))
    701 if result is None and exc is None:
--> 702     raise ve_exc
    703 elif exc is None:
    704     exc = TypeError(
    705             'error in %s._missing_: returned %r instead of None or a valid member'
    706             % (cls.__name__, result)
    707             )

ValueError: 'Penguin' is not a valid ObsMode
```

FWIW you can do perverse things like
```
In [7]: ObsMode.Volt in ObsMode
Out[7]: True
```

idk, but sometimes I get tired of using quotes and stringly typed data get on my nerves